config/bindings/namespaced/auth_proxy_role_binding.yaml

```diff
@@ -0,0 +1,12 @@
+apiVersion: rbac.authorization.k8s.io/v1
```

Comment:
Why are these `proxy-role`, `proxy-rolebinding`, and `manager-rolebinding` duplicated between namespace and clusterscope targets? We should not duplicate these lest we want to invite bugs when we update one and not the other...

Reply:
I don't know of a good way to remove code duplication here. All the documentation about overlays only modifies the spec of the resources, relying on the Group and Kind as the selectors. We would like to modify the Group and Kind. The only real way to do this is the messy JSON patches. I did it this way before but it made for a lot of fairly unintelligible code and broke the resource resolution that Kustomize does behind the scenes.